src/reducers/visitorReducer.js

```diff
@@ -0,0 +1,28 @@
+import { CHANGE_NAME } from '../actions/changeNameAction';
+import { CLEAR_FAVORITES, UPDATE_FAVORITE } from '../actions/favoriteActions';
+
+const initialState = {
+  favorites: {},
+  name: ''
+};
+
+export default (state = initialState, action) => {
+  const { 
+    type, 
+    favoriteType,
+    favoriteValue,
+    name
+  } = action;
+
+  const newState = { ...state };
+  newState.name = type === CHANGE_NAME ? name : state.name;
+
+  if (type === UPDATE_FAVORITE) {
+    newState.favorites[favoriteType] = favoriteValue;
```

Comment:
Since this data is local to the scope of this function, I think you won't get burned for modifying the data. This jumped out at me as weird though.

It might be fun to try Immutable.js at some point to see if you can avoid mutations: https://github.com/facebook/immutable-js.

Reply:
Would it be better to have something like 
```
const newFavorites = { ...state.favorites };
newFavorites[favoriteType] = favoriteValue;
return { ...state, favorites}
```

Reply:
The issue is the assignment operator. So both examples have the issue. Again, I don't think it matters *too* much since it's local state and not global state, but it's still a little odd. 

A potential way around using the assignment operator could be to use `Object.assign` or the object spread operator `...` on the current state's `favorites`.

Reply:
I don't know how to do that since `favoriteType` is a variable whose value is what I want to use as the key.  A way Brian M came up with to avoid it is to have the action pass in an object containing the new favorite and then use spread to merge the incoming object with the existing favorites in state.

Reply:
If I understand it right, it could be done like this:

```
const newFavorites = Object.assign(state.favorites, { [favoriteType]: favoriteValue });
return {
   ...state,
  favorites: newFavorites
}
```